diff --git a/node_modules/bigeval/BigEval.js b/node_modules/bigeval/BigEval.js
index 45253bd..65717ef 100644
--- a/node_modules/bigeval/BigEval.js
+++ b/node_modules/bigeval/BigEval.js
@@ -4,12 +4,10 @@
 	https://github.com/aviaryan/bigEval.js
 */
 
-var root = Function('return this')();
-
 /**
  * @enum {string}
  * @readonly
-*/
+ */
 var TokenType = {
 	STRING: '"',
 	VAR: 'x',
@@ -22,25 +20,25 @@ var TokenType = {
 	COMMA: ','
 };
 
-var hasOwnProperty = Object.hasOwnProperty;
-
 var DEFAULT_VAR_NAME_CHARS = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_$';
 
-var BigEval = function() {
+var BigEval = function(root = Function('return this')()) {
+
+	this.root = root;
 
 	// https://en.wikipedia.org/wiki/Order_of_operations#Programming_languages
 
 	this.order = [
-				['!'],
-				['**'],
-				['\\', '/', '*', '%'],
-				['+', '-'],
-				['<<', '>>'],
-				['<', '<=', '>', '>='],
-				['==', '=', '!=', '<>'],
-				['&'], ['^'], ['|'],
-				['&&'], ['||']
-				];
+		['!'],
+		['**'],
+		['\\', '/', '*', '%'],
+		['+', '-'],
+		['<<', '>>'],
+		['<', '<=', '>', '>='],
+		['==', '=', '!=', '<>'],
+		['&'], ['^'], ['|'],
+		['&&'], ['||']
+	];
 
 	this.prefixOps = ['!'];
 	this.suffixOps = ['!'];
@@ -568,7 +566,7 @@ BigEval.prototype.compile = function (expression) {
 			prevToken.type === TokenType.OP &&
 			(prevToken.value === '-' || prevToken.value === '+') &&
 			((i > 1 && tokens[i - 2].type === TokenType.OP) || i === 1)
-			) {
+		) {
 
 			if (prevToken.value === '-') {
 				token.value = prevToken.value + token.value;
@@ -729,8 +727,8 @@ BigEval.prototype._evaluateFunction = function (token) {
 	else if (typeof(Math[fname]) == 'function') {
 		return Math[fname].apply(Math, args);
 	}
-	else if (typeof(root[fname]) == 'function') {
-		return root[fname].apply(root, args);
+	else if (typeof(this.root[fname]) == 'function') {
+		return this.root[fname].apply(this.root, args);
 	}
 
 	throw new Error('Function named "' + fname + '" was not found');
